### ISSUE TITLE
#include "ios" error in mac Mojave

### ISSUE THREAD

**Issue by kyoungrok0517** [huggingface/neuralcoref#97]
Hi, I encounter the following error when I try to install the models

```bash
Processing ./en_coref_sm-3.0.0.tar.gz
Requirement already satisfied: spacy>=>=2.0.0a18 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from en-coref-sm==3.0.0) (2.0.12)
Requirement already satisfied: numpy>=1.7 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (1.15.2)
Collecting regex==2017.4.5 (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0)
Requirement already satisfied: requests<3.0.0,>=2.13.0 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (2.18.4)
Requirement already satisfied: preshed<2.0.0,>=1.0.0 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (1.0.0)
Requirement already satisfied: murmurhash<0.29,>=0.28 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (0.28.0)
Requirement already satisfied: plac<1.0.0,>=0.9.6 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (0.9.6)
Requirement already satisfied: ujson>=1.35 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (1.35)
Requirement already satisfied: cymem<1.32,>=1.30 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (1.31.2)
Requirement already satisfied: dill<0.3,>=0.2 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (0.2.7.1)
Requirement already satisfied: thinc<6.11.0,>=6.10.3 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (6.10.3)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from requests<3.0.0,>=2.13.0->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (3.0.4)
Requirement already satisfied: idna<2.7,>=2.5 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from requests<3.0.0,>=2.13.0->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (2.6)
Requirement already satisfied: urllib3<1.23,>=1.21.1 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from requests<3.0.0,>=2.13.0->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (1.22)
Requirement already satisfied: certifi>=2017.4.17 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from requests<3.0.0,>=2.13.0->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (2018.8.24)
Requirement already satisfied: six<2.0.0,>=1.10.0 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from thinc<6.11.0,>=6.10.3->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (1.11.0)
Requirement already satisfied: cytoolz<0.10,>=0.9.0 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from thinc<6.11.0,>=6.10.3->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (0.9.0.1)
Requirement already satisfied: tqdm<5.0.0,>=4.10.0 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from thinc<6.11.0,>=6.10.3->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (4.22.0)
Requirement already satisfied: msgpack-numpy<1.0.0,>=0.4.1 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from thinc<6.11.0,>=6.10.3->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (0.4.1)
Requirement already satisfied: wrapt<1.11.0,>=1.10.0 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from thinc<6.11.0,>=6.10.3->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (1.10.11)
Requirement already satisfied: msgpack<1.0.0,>=0.5.6 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from thinc<6.11.0,>=6.10.3->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (0.5.6)
Requirement already satisfied: toolz>=0.8.0 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from cytoolz<0.10,>=0.9.0->thinc<6.11.0,>=6.10.3->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (0.9.0)
Requirement already satisfied: msgpack-python>=0.3.0 in /Users/kyoungrok/anaconda/lib/python3.6/site-packages (from msgpack-numpy<1.0.0,>=0.4.1->thinc<6.11.0,>=6.10.3->spacy>=>=2.0.0a18->en-coref-sm==3.0.0) (0.5.4)
Building wheels for collected packages: en-coref-sm
  Running setup.py bdist_wheel for en-coref-sm ... error
  Complete output from command /Users/kyoungrok/anaconda/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/8n/v5p4940n2xbcn7svgbrqb8zh0000gn/T/pip-req-build-cor5g7o5/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /private/var/folders/8n/v5p4940n2xbcn7svgbrqb8zh0000gn/T/pip-wheel-ww8axwrm --python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.7-x86_64-3.6
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm
  copying en_coref_sm/__init__.py -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/neuralcoref
  copying en_coref_sm/neuralcoref/__init__.py -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/neuralcoref
  copying en_coref_sm/__init__.pxd -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0
  copying en_coref_sm/en_coref_sm-3.0.0/tokenizer -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0
  copying en_coref_sm/en_coref_sm-3.0.0/meta.json -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/ner
  copying en_coref_sm/en_coref_sm-3.0.0/ner/lower_model -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/ner
  copying en_coref_sm/en_coref_sm-3.0.0/ner/moves -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/ner
  copying en_coref_sm/en_coref_sm-3.0.0/ner/cfg -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/ner
  copying en_coref_sm/en_coref_sm-3.0.0/ner/tok2vec_model -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/ner
  copying en_coref_sm/en_coref_sm-3.0.0/ner/upper_model -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/ner
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/parser
  copying en_coref_sm/en_coref_sm-3.0.0/parser/lower_model -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/parser
  copying en_coref_sm/en_coref_sm-3.0.0/parser/moves -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/parser
  copying en_coref_sm/en_coref_sm-3.0.0/parser/cfg -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/parser
  copying en_coref_sm/en_coref_sm-3.0.0/parser/tok2vec_model -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/parser
  copying en_coref_sm/en_coref_sm-3.0.0/parser/upper_model -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/parser
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/vocab
  copying en_coref_sm/en_coref_sm-3.0.0/vocab/vectors -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/vocab
  copying en_coref_sm/en_coref_sm-3.0.0/vocab/lexemes.bin -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/vocab
  copying en_coref_sm/en_coref_sm-3.0.0/vocab/strings.json -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/vocab
  copying en_coref_sm/en_coref_sm-3.0.0/vocab/key2row -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/vocab
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref
  copying en_coref_sm/en_coref_sm-3.0.0/neuralcoref/cfg -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref
  copying en_coref_sm/en_coref_sm-3.0.0/neuralcoref/single_model -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref
  copying en_coref_sm/en_coref_sm-3.0.0/neuralcoref/pairs_model -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref/static_vectors
  copying en_coref_sm/en_coref_sm-3.0.0/neuralcoref/static_vectors/vectors -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref/static_vectors
  copying en_coref_sm/en_coref_sm-3.0.0/neuralcoref/static_vectors/key2row -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref/static_vectors
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref/tuned_vectors
  copying en_coref_sm/en_coref_sm-3.0.0/neuralcoref/tuned_vectors/vectors -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref/tuned_vectors
  copying en_coref_sm/en_coref_sm-3.0.0/neuralcoref/tuned_vectors/key2row -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/neuralcoref/tuned_vectors
  creating build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/tagger
  copying en_coref_sm/en_coref_sm-3.0.0/tagger/tag_map -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/tagger
  copying en_coref_sm/en_coref_sm-3.0.0/tagger/cfg -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/tagger
  copying en_coref_sm/en_coref_sm-3.0.0/tagger/model -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/en_coref_sm-3.0.0/tagger
  copying en_coref_sm/meta.json -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm
  copying en_coref_sm/neuralcoref/neuralcoref.pyx -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/neuralcoref
  copying en_coref_sm/neuralcoref/__init__.pxd -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/neuralcoref
  copying en_coref_sm/neuralcoref/neuralcoref.pxd -> build/lib.macosx-10.7-x86_64-3.6/en_coref_sm/neuralcoref
  running build_ext
  building 'en_coref_sm.neuralcoref.neuralcoref' extension
  creating build/temp.macosx-10.7-x86_64-3.6
  creating build/temp.macosx-10.7-x86_64-3.6/en_coref_sm
  creating build/temp.macosx-10.7-x86_64-3.6/en_coref_sm/neuralcoref
  gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/kyoungrok/anaconda/include -arch x86_64 -I/Users/kyoungrok/anaconda/include -arch x86_64 -I/Users/kyoungrok/anaconda/include/python3.6m -I/private/var/folders/8n/v5p4940n2xbcn7svgbrqb8zh0000gn/T/pip-req-build-cor5g7o5/include -I/Users/kyoungrok/anaconda/include/python3.6m -c en_coref_sm/neuralcoref/neuralcoref.cpp -o build/temp.macosx-10.7-x86_64-3.6/en_coref_sm/neuralcoref/neuralcoref.o
  warning: include path for stdlibc++ headers not found; pass '-std=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
  In file included from en_coref_sm/neuralcoref/neuralcoref.cpp:580:
  In file included from /private/var/folders/8n/v5p4940n2xbcn7svgbrqb8zh0000gn/T/pip-req-build-cor5g7o5/include/numpy/arrayobject.h:15:
  In file included from /private/var/folders/8n/v5p4940n2xbcn7svgbrqb8zh0000gn/T/pip-req-build-cor5g7o5/include/numpy/ndarrayobject.h:17:
  In file included from /private/var/folders/8n/v5p4940n2xbcn7svgbrqb8zh0000gn/T/pip-req-build-cor5g7o5/include/numpy/ndarraytypes.h:1728:
  /private/var/folders/8n/v5p4940n2xbcn7svgbrqb8zh0000gn/T/pip-req-build-cor5g7o5/include/numpy/npy_deprecated_api.h:11:2: warning: "Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
  #warning "Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION"
   ^
  en_coref_sm/neuralcoref/neuralcoref.cpp:583:10: fatal error: 'ios' file not found
  #include "ios"
           ^~~~~
  2 warnings and 1 error generated.
  error: command 'gcc' failed with exit status 1

  ----------------------------------------
  Failed building wheel for en-coref-sm
```

**Comment by EricCacciavillani** [comment]
Having the same problem here with #include "ios"...

**Comment by nityansuman** [comment]
Having exact same issue!

**Comment by leong1016** [comment]
Same issue on the new Mac version.

**Comment by ecarlso2verizon** [comment]
Try setting the CFLAGS stdlib=libc++. CFLAGS='-stdlib=libc++' 
This worked for me.

**Comment by jmugan** [comment]
Same issue here. @ecarlson6022, can you give a little more detail?

**Comment by ecarlson6022** [comment]
@jmugan when you run the pip install command add CFLAGS='-stdlib=libc++' in front. 
CFLAGS='-stdlib=libc++' pip install MODEL_URL

**Comment by jmugan** [comment]
Since my earlier post, I've updated my mac, and the comment of @ecarlson6022, which used to work, no longer does. I get the error `clang: error: invalid deployment target for -stdlib=libc++ (requires OS X 10.7 or later)`. Any suggestions?

**Comment by thomwolf** [comment]
Hi all, I am having a look at that right now.
I guess it's something like that: https://github.com/cython/cython/issues/2694

**Comment by JohnGiorgi** [comment]
The following worked for me on MacOS Mojave:

```
CFLAGS='-stdlib=libc++' pip install spacy==2.0.12 thinc==6.10.3 google-compute-engine https://github.com/huggingface/neuralcoref-models/releases/download/en_coref_md-3.0.0/en_coref_md-3.0.0.tar.gz
```

Unfortunately, I don't know why. Just pieced this together from different issues.

**Comment by emanuel1025** [comment]
> Since my earlier post, I've updated my mac, and the comment of @ecarlson6022, which used to work, no longer does. I get the error `clang: error: invalid deployment target for -stdlib=libc++ (requires OS X 10.7 or later)`. Any suggestions?

Same issue.

**Comment by RasmusIven** [comment]
solution presented by @ecarlson6022 and @JohnGiorg don't do it for me, I still recieve a fatal error: 'ios' file not found #include "ios"

**Comment by Abhishek-Arora** [comment]
> The following worked for me on MacOS Mojave:
> 
> ```
> CFLAGS='-stdlib=libc++' pip install spacy==2.0.12 thinc==6.10.3 google-compute-engine https://github.com/huggingface/neuralcoref-models/releases/download/en_coref_md-3.0.0/en_coref_md-3.0.0.tar.gz
> ```
> Unfortunately, I don't know why. Just pieced this together from different issues.

Thank you! This worked for me.

**Comment by thomwolf** [comment]
All these installation errors should be fixed in the new release (4.0).
Installation can now be simply done with pip (see the README for details).
